pkg/provider/kubernetes/gateway/kubernetes.go

```diff
@@ -463,29 +467,41 @@ func (p *Provider) fillGatewayConf(client Client, gateway *v1alpha1.Gateway, con
 				}
 
 				if routeRule.ForwardTo != nil {
-					wrrService, subServices, err := loadServices(client, gateway.Namespace, routeRule.ForwardTo)
-					if err != nil {
-						// update "ResolvedRefs" status true with "DroppedRoutes" reason
-						listenerStatuses[i].Conditions = append(listenerStatuses[i].Conditions, metav1.Condition{
-							Type:               string(v1alpha1.ListenerConditionResolvedRefs),
-							Status:             metav1.ConditionFalse,
-							LastTransitionTime: metav1.Now(),
-							Reason:             string(v1alpha1.ListenerReasonDegradedRoutes),
-							Message:            fmt.Sprintf("Cannot load service from HTTPRoute %s/%s : %v", gateway.Namespace, httpRoute.Name, err),
-						})
-
-						// TODO update the RouteStatus condition / deduplicate conditions on listener
-						continue
-					}
+					// Traefik internal service can be used only if there is only one ForwardTo service reference.
+					isInternalService := len(routeRule.ForwardTo) == 1 &&
+						routeRule.ForwardTo[0].ServiceName == nil &&
+						routeRule.ForwardTo[0].BackendRef != nil &&
+						routeRule.ForwardTo[0].BackendRef.Kind == traefikServiceKind &&
+						routeRule.ForwardTo[0].BackendRef.Group == traefikServiceGroupName &&
+						strings.HasSuffix(routeRule.ForwardTo[0].BackendRef.Name, "@internal")
+
+					if isInternalService {
+						router.Service = routeRule.ForwardTo[0].BackendRef.Name
+					} else {
```

Comment:
could this be moved to its own function so that we could do:

```
if isInternalService(routeRule.ForwardTo) {}
```

It would also allow that to be testable, and could simplify this logic.

Reply:
I agree with you, but I think the condition should be:
```
if len(routeRule.ForwardTo) == 1 && isInternalService(routeRule.ForwardTo) {
...
}
```

The fact that we check it has only one element in `ForwardTo` is purely a limitation we impose on the WRR and not a feature of an internal service.